devicemap-schema.json

```diff
@@ -0,0 +1,145 @@
+{
+    "type": "object",
+    "properties": {
+        "clei": {
+            "type": "string"
+        },
+        "part_number": {
+            "type": "string"
+        },
+        "ethernet": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "allOf": [
+                    {
+                        "$ref": "#/$defs/baseInterface"
+                    }
+                ],
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "pattern": "^(ge|xe|mgmt)-[0-9]+(-[0-9]+)?$"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": [
+                            "WAN",
+                            "LAN",
+                            "HASync",
+                            "HAFabric",
+                            "MGMT"
+                        ]
+                    },
+                    "pciAddress": {
+                        "type": [
+                            "string",
+                            "null"
+                        ],
+                        "pattern": "^[0-9a-fA-F]{4}(:[0-9a-fA-F]{2}){2}\\.[0-7]$"
+                    },
+                    "vmbusId": {
+                        "type": [
+                            "string",
+                            "null"
+                        ],
+                        "pattern": "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}"
+                    },
+                    "speed": {
+                        "type": "integer"
+                    }
+                },
+                "required": [
+                    "name",
```

Comment:
I think this section might need some updated, no? the `pciAddress` and `vmbusId` are mutually exclusive but you need at least one of them.

Reply:
I'll rework this section. _technically_ this should pass validation because we always write `pciAddress/vmbusId: null` but it should definitely not work on a technicality